<a href="https://colab.research.google.com/github/jjaquezrd/Grupo_2_Data_Science/blob/main/EjercicioNike.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [22]:
df = pd.read_csv('/content/drive/MyDrive/Data_Science/nike_data_2022_09.csv')
df.head()

index                                                url  \
0      0  https://www.nike.com/t/dri-fit-team-minnesota-...   
1      1  https://www.nike.com/t/club-américa-womens-dri...   
2      2  https://www.nike.com/t/sportswear-swoosh-mens-...   
3      3  https://www.nike.com/t/dri-fit-one-luxe-big-ki...   
4      4  https://www.nike.com/t/paris-saint-germain-rep...   

                                      name  \
0  Nike Dri-FIT Team (MLB Minnesota Twins)   
1                             Club América   
2                   Nike Sportswear Swoosh   
3                    Nike Dri-FIT One Luxe   
4    Paris Saint-Germain Repel Academy AWF   

                                           sub_title brand     model  \
0                          Men's Long-Sleeve T-Shirt  Nike  14226571   
1           Women's Nike Dri-FIT Soccer Jersey Dress  Nike  13814665   
2                                     Men's Overalls  Nike  13015648   
3  Big Kids' (Girls') Printed Tights (Extended Size)  Nike  13809796   
4                            Big Kids' Soccer Jacket  Nike  13327415   

                                 color   price currency availability  \
0                                 Navy   40.00      USD      InStock   
1                          Black/Black   90.00      USD      InStock   
2                          Black/White  140.00      USD   OutOfStock   
3                      Black/Rush Pink   22.97      USD   OutOfStock   
4  Dark Grey/Black/Siren Red/Siren Red   70.00      USD      InStock   

                                         description  \
0  SWEAT-WICKING COMFORT.The Nike Dri-FIT Team (M...   
1  Inspired by traditional soccer jerseys, the Cl...   
2  WORKING HARD TO KEEP YOU COMFORTABLE.The Nike ...   
3  ELEVATED COMFORT GOES FULL BLOOM.The Nike Dri-...   
4  WATER-REPELLENT COVERAGE GETS PSG DETAILS.The ...   

                                     raw_description  avg_rating  \
0  <div class="pi-pdpmainbody"><p><b class="headl...         NaN   
1  <div class="pi-pdpmainbody"><br/><p>Inspired b...         5.0   
2  <div class="pi-pdpmainbody"><p><b class="headl...         4.9   
3  <div class="pi-pdpmainbody"><p><b class="headl...         NaN   
4  <div class="pi-pdpmainbody"><p><b class="headl...         NaN   

   review_count                                             images  \
0           NaN  https://static.nike.com/a/images/t_PDP_1280_v1...   
1           1.0  https://static.nike.com/a/images/t_PDP_1280_v1...   
2          11.0  https://static.nike.com/a/images/t_PDP_1280_v1...   
3           NaN  https://static.nike.com/a/images/t_PDP_1280_v1...   
4           NaN  https://static.nike.com/a/images/t_PDP_1280_v1...   

        available_sizes                               uniq_id  \
0  S | M | L | XL | 2XL  c3229e54-aa58-5fdd-9f71-fbe66366b2b2   
1             L (12–14)  f8ebb2ed-17ae-5719-b750-5ea3ec69b75c   
2                   NaN  88120081-e6cb-5399-b9dc-a2d3d5dd5206   
3                   NaN  98348cc5-1520-5b6e-a5f6-c42547b6a092   
4   XS | S | M | L | XL  f15981a5-d8c9-53fa-880d-80606be188fe   

            scraped_at  
0  20/09/2022 23:32:28  
1  20/09/2022 23:32:40  
2  20/09/2022 23:33:16  
3  20/09/2022 23:33:17  
4  20/09/2022 23:33:22

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112 entries, 0 to 111
Data columns (total 18 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   index            112 non-null    int64  
 1   url              112 non-null    object 
 2   name             112 non-null    object 
 3   sub_title        112 non-null    object 
 4   brand            112 non-null    object 
 5   model            112 non-null    int64  
 6   color            110 non-null    object 
 7   price            112 non-null    float64
 8   currency         112 non-null    object 
 9   availability     108 non-null    object 
 10  description      112 non-null    object 
 11  raw_description  112 non-null    object 
 12  avg_rating       23 non-null     float64
 13  review_count     23 non-null     float64
 14  images           108 non-null    object 
 15  available_sizes  56 non-null     object 
 16  uniq_id          112 non-null    object 
 17  scraped_at      

## Eliminar columnas no necesarias para el analisis


In [24]:
## Raw description
## images
df= df.drop(['raw_description','images'], axis=1)

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112 entries, 0 to 111
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   index            112 non-null    int64  
 1   url              112 non-null    object 
 2   name             112 non-null    object 
 3   sub_title        112 non-null    object 
 4   brand            112 non-null    object 
 5   model            112 non-null    int64  
 6   color            110 non-null    object 
 7   price            112 non-null    float64
 8   currency         112 non-null    object 
 9   availability     108 non-null    object 
 10  description      112 non-null    object 
 11  avg_rating       23 non-null     float64
 12  review_count     23 non-null     float64
 13  available_sizes  56 non-null     object 
 14  uniq_id          112 non-null    object 
 15  scraped_at       112 non-null    object 
dtypes: float64(3), int64(2), object(11)
memory usage: 14.1+ KB


Cambiando el tipo de datos de scraped_at a fecha a fin de verificar si la extraccion de datos fue en un unico dia o encierra algun periodo de tiempo

In [26]:
df['scraped_at']= pd.to_datetime(df['scraped_at']).dt.date

In [31]:
df.groupby('scraped_at')['index'].count()

scraped_at
2022-09-20    112
Name: index, dtype: int64

Se confrima que la extraccion de datos fue realizada en la misma fecha. No se contiene un rango de tiempo a fin de construir un historico o visualizar tendencias

Examen sobre la columna 'Avialiability' ya que presenta solo 4 registros nulos, por lo que existe posibilidad de eliminarla. 

In [36]:
aviability_fileter = df['availability'].isna()
df[aviability_fileter].head()

index                                                url  \
12     12  https://www.nike.com/t/croatia-2022-23-stadium...   
17     17  https://www.nike.com/t/portugal-2022-23-stadiu...   
29     29  https://www.nike.com/t/dri-fit-crucial-catch-b...   
92     92  https://www.nike.com/t/us-academy-pro-big-kids...   

                                              name  \
12                    Croatia 2022/23 Stadium Away   
17                   Portugal 2022/23 Stadium Home   
29  Nike Dri-FIT Crucial Catch (NFL Buffalo Bills)   
92                                U.S. Academy Pro   

                                         sub_title brand     model  \
12              Women's Nike Dri-FIT Soccer Jersey  Nike  14063839   
17                Men's Nike Dri-FIT Soccer Shorts  Nike  14063778   
29                         Women's Pullover Hoodie  Nike  14234141   
92  Big Kids' Nike Dri-FIT Short-Sleeve Soccer Top  Nike  14062977   

                               color  price currency availability  \
12     Blackened Blue/University Red   95.0      USD          NaN   
17  Gorge Green/Pepper Red/Gold Dart   52.0      USD          NaN   
29                             Black   85.0      USD          NaN   
92                    Obsidian/White   30.0      USD          NaN   

                                          description  avg_rating  \
12   Like other jerseys from our Stadium collectio...         NaN   
17   Like other shorts from our Stadium collection...         NaN   
29  LIGHTWEIGHT COVERAGE.The Nike Dri-FIT Crucial ...         NaN   
92  UP YOUR GAME.After conquering the basics, take...         NaN   

    review_count available_sizes                               uniq_id  \
12           NaN             NaN  4a3f7333-1945-5f1e-b0e3-829f40f28bb3   
17           NaN             NaN  c3fc0349-a9e4-5610-a676-9f7b5e8b4dfd   
29           NaN             NaN  dc43a48d-3767-5991-9608-f5ef49c4c9d8   
92           NaN             NaN  78db57cc-f5a2-5542-8400-ab327e91cfef   

    scraped_at  
12  2022-09-20  
17  2022-09-20  
29  2022-09-20  
92  2022-09-20

Se confirma que los cuatro registros para los cuales no se registra la disponibilidad, segun su descripcion corresponden a versiones limitadas de eventos o equipos deportivos. Se recomienda eliminacion

In [48]:
indexna= df[aviability_fileter].index
df.drop(indexna, inplace=True)

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 108 entries, 0 to 111
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   index            108 non-null    int64  
 1   url              108 non-null    object 
 2   name             108 non-null    object 
 3   sub_title        108 non-null    object 
 4   brand            108 non-null    object 
 5   model            108 non-null    int64  
 6   color            106 non-null    object 
 7   price            108 non-null    float64
 8   currency         108 non-null    object 
 9   availability     108 non-null    object 
 10  description      108 non-null    object 
 11  avg_rating       23 non-null     float64
 12  review_count     23 non-null     float64
 13  available_sizes  56 non-null     object 
 14  uniq_id          108 non-null    object 
 15  scraped_at       108 non-null    object 
dtypes: float64(3), int64(2), object(11)
memory usage: 14.3+ KB


## Analisis valores nulos columnas Avg_rating y Review_count
Estas columnas presentan un alto indice de datos nulos, siendo estas columnas reelevantes para un analisis de predicicion de compras y ventas.

In [56]:
##Filtro para verificar la relacion entre los nulos de la columna avg_rating y review_count. Determinar
#si cuando una columna esta vacia la otra tambien.
filtroNa = df['avg_rating'].isna() & df['review_count'].isna()
df[filtroNa].groupby(['name','sub_title'])['index'].count()

name                                       sub_title                           
Air Jordan 1 Low SE                        Big Kids' Shoes                         1
Air Jordan XXXV Low DS PF                  Basketball Shoes                        1
Air Jordan XXXVI Low Luka PF               Men's Basketball Shoes                  1
Charlotte Hornets                          Men's Jordan NBA T-Shirt                1
Chelsea FC Repel Academy AWF               Men's Soccer Jacket                     1
                                                                                  ..
Paris-Saint Germain 2021/22 Stadium Third  Big Kids' Nike Dri-FIT Soccer Jersey    1
Philadelphia 76ers                         Men's Nike NBA Shorts                   1
Sacramento Kings Icon Edition              Men's Nike NBA Swingman Shorts          1
Sylvia Fowles Lynx Explorer Edition        Nike Dri-FIT WNBA Victory Jersey        1
Team 31                                    Women's Nike NBA Cropped T-

Se confirma la relacion entre las columnas bajo analisis ya que 85 registros de 108 poseen ambas columnas vacias. Los articulos que presentan estas columnas vacias se destacan en los grupos arriba.